backend/tracim_backend/views/core_api/user_controller.py

```diff
@@ -603,6 +605,48 @@ def get_user_messages(
             request.candidate_user.user_id, ReadStatus(hapic_data.query["read_status"])
         )
 
+    @hapic.with_api_doc(tags=[SWAGGER_TAG__USER_EVENT_ENDPOINTS])
+    @check_right(has_personal_access)
+    @hapic.input_path(UserIdPathSchema())
+    @hapic.output_body(NoContentSchema(), default_http_code=HTTPStatus.NO_CONTENT)
+    def read_all_messages(self, context, request: TracimRequest, hapic_data: HapicData) -> None:
+        """
+        Read all unread message for user
+        """
+        app_config = request.registry.settings["CFG"]  # type: CFG
+        event_api = EventApi(request.current_user, request.dbsession, app_config)
+        event_api.read_all_messages_of_user(request.candidate_user.user_id)
+
+    @hapic.with_api_doc(tags=[SWAGGER_TAG__USER_EVENT_ENDPOINTS])
+    @hapic.handle_exception(MessageDoesNotExist, http_code=HTTPStatus.BAD_REQUEST)
```

Comment:
Is this case really a bad request? or only a not found?

Reply:
According to current usage, yes, we do use bad request for other case where parameter given doesn't match with existing elements. The error_code explain the exact reason
404 not found is used to mean "the endpoint doesn't exist".